src/util/interaction/out.ts

```diff
@@ -36,6 +36,8 @@ export function progress<T>(title: string, action: Promise<T>): Promise<T> {
 // function.
 //
 export function list<T>(formatter: {(item: T): string}, items: T[]): void {
+  if (!items || Object.keys(items).length === 0) { return; }
```

Comment:
This check seems a little weird for a function that's supposed to take an array - why are you checking `Object.keys(items).length`?